addons/sale/migrations/9.0.1.0/pre-migration.py

```diff
@@ -56,9 +56,85 @@ def map_order_state(cr):
         WHERE sol.order_id = so.id""")
 
 
+def prepopulate_fields(cr):
+    """ Recomputing a fields will be very expensive via the ORM, so do it
+    here for fields where the computation is trivial"""
+    cr.execute('alter table sale_order_line add column qty_invoiced numeric')
+    cr.execute(
+        """\
+        with line2qty_invoiced(id, qty_invoiced) as (
+            select l.id, sum(
+                case
+                    when i.type='out_invoice' then il.quantity
+                    when i.type='out_refund' then -il.quantity
+                end * uom_invoice.factor / uom_sale.factor
+            ) s
+            from sale_order_line l
+            join sale_order_line_invoice_rel r on r.order_line_id=l.id
+            join account_invoice_line il on r.invoice_line_id=il.id
+            join account_invoice i on i.id=il.invoice_id
+            join product_uom uom_invoice on il.uom_id=uom_invoice.id
+            join product_uom uom_sale on l.product_uom=uom_sale.id
+            group by l.id
+        )
+        update sale_order_line
+        set qty_invoiced=line2qty_invoiced.qty_invoiced
+        from line2qty_invoiced where line2qty_invoiced.id=sale_order_line.id
+        """)
+    cr.execute('alter table sale_order_line add column qty_to_invoice numeric')
```

Comment:
ADD `set default=product_uom_qty`

Reply:
About this commit, PG says: cannot use column references in default expression

Reply:
From PG doc:

> DEFAULT default_expr

> The DEFAULT clause assigns a default data value for the column whose column definition it appears within. The value is any variable-free expression (subqueries and cross-references to other columns in the current table are not allowed). The data type of the default expression must match the data type of the column.

> The default expression will be used in any insert operation that does not specify a value for the column. If there is no default for a column, then the default is null.

Reply:
OK, I didn't know. Then we should still keep here 2 sentences

Reply:
I'm working on https://github.com/hbrunn/OpenUpgrade/pull/5.  I hope next week this can be merged.